File: 228. Summary Ranges/Solution228.py
class Solution(object):
    def summaryRanges(self, nums):
        ans = []
        i = 0

        while i < len(nums):
            start = nums[i]

            while i < len(nums) - 1 and nums[i] + 1 == nums[i + 1]:
                i += 1

            if start != nums[i]:
                ans.append(str(start) + "->" + str(nums[i]))
            else:
                ans.append(str(nums[i]))    
            i += 1
            
        return ans

    

c = Solution()
print(c.summaryRanges([0,1,2,4,5,7]))

File: 228. Summary Ranges/Solution228.java

import java.util.ArrayList;
import java.util.List;

class Solution228 {
    public static List<String> summaryRanges(int[] nums, ArrayList<String> arrayList) {
        ArrayList<String> list = new ArrayList<>();
        int i = 0;
        while (i < nums.length) {
            int start = nums[i];
            while (i < nums.length - 1 && nums[i] + 1 == nums[i + 1]) {
                i++;
            }
            if (start != nums[i]) {
                list.add(start + "->" + nums[i]);
            } else {
                list.add(String.valueOf(nums[i]));
            }
            i++;
        }

        return list;

    }

    public static void main(String[] args) {
        int[] nums = { 0, 1, 2, 4, 5, 7 };
        System.out.println(summaryRanges(nums, new ArrayList<String>()));
    }
}